File: app/src/main/java/tech/diarmaid/koohiiaite/widget/KanjiSearchView.java
package tech.diarmaid.koohiiaite.widget;

import android.content.Context;
import android.support.v7.widget.SearchView;
import android.util.AttributeSet;

/**
 * Override the search view so that we don't clear the search
 * when the "up" button is pressed
 */
public class KanjiSearchView extends SearchView {

    public KanjiSearchView(Context context) {
        super(context);
    }

    public KanjiSearchView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public KanjiSearchView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    public void onActionViewCollapsed() {
        CharSequence query = getQuery();
        super.onActionViewCollapsed();
        setQuery(query, false);
    }

    @Override
    public void onActionViewExpanded() {
        CharSequence query = getQuery();
        super.onActionViewExpanded();
        setQuery(query, false);
    }
}

File: app/src/main/java/tech/diarmaid/koohiiaite/widget/KanjiSearchView.kt
package tech.diarmaid.koohiiaite.widget

import android.content.Context
import android.support.v7.widget.SearchView
import android.util.AttributeSet

/**
 * Override the search view so that we don't clear the search
 * when the "up" button is pressed
 */
class KanjiSearchView : SearchView {

    constructor(context: Context) : super(context)

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs)

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr)

    override fun onActionViewCollapsed() {
        val query = query
        super.onActionViewCollapsed()
        setQuery(query, false)
    }

    override fun onActionViewExpanded() {
        val query = query
        super.onActionViewExpanded()
        setQuery(query, false)
    }
}
